src/main/java/org/aksw/simba/squirrel/sink/impl/sparql/SparqlBasedSink.java

```diff
@@ -0,0 +1,82 @@
+package org.aksw.simba.squirrel.sink.impl.sparql;
+
+import org.aksw.simba.squirrel.data.uri.CrawleableUri;
+import org.aksw.simba.squirrel.sink.Sink;
+import org.apache.jena.graph.Triple;
+import org.apache.jena.update.UpdateExecutionFactory;
+import org.apache.jena.update.UpdateFactory;
+import org.apache.jena.update.UpdateProcessor;
+import org.apache.jena.update.UpdateRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class SparqlBasedSink implements Sink {
+
+    public static final String JENA_PORTS_KEY = "JENA_PORT";
+    public static final String JENA_CONTAINER_NAME_KEY = "JENA_HOST_NAME";
+    private static final int SENDING_INTERVAL_BUFFERED_LIST = 100;
+    private static String strContentDatasetUriUpdate;
+    private static String strContentDatasetUriQuery;
+    private List<Triple> lstBufferedTriples;
+
+    @SuppressWarnings("unused")
+    private static final Logger LOGGER = LoggerFactory.getLogger(SparqlBasedSink.class);
+
+
+    public SparqlBasedSink() throws Exception {
+        Map<String, String> env = System.getenv();
+        String containerName = null;
+        String port = null;
+        if (env.containsKey(JENA_CONTAINER_NAME_KEY) || env.containsKey(JENA_PORTS_KEY)) {
+            containerName = env.get(JENA_CONTAINER_NAME_KEY);
+            port = env.get(JENA_PORTS_KEY);
+        } else {
+            String msg = "Couldn't get " + JENA_CONTAINER_NAME_KEY + " or " + JENA_PORTS_KEY + " from the environment.";
+            throw new Exception(msg);
+        }
+
+        String datasetPrefix = "http://" + containerName + ":" + port + "/ContentSet/";
+        strContentDatasetUriUpdate = datasetPrefix + "update";
+        strContentDatasetUriQuery = datasetPrefix + "query";
+    }
```

Comment:
Although this might be working, it is focusing very much on the current use case. 

1. Please provide a constructor that accepts the two necessary Strings, i.e., the query and the update URL.
2. Please rename the "JENA" parts to something like "SPARQL_HOST" or something similar :wink:

Reply:
fixed :+1: